source/common/listener_manager/active_tcp_socket.cc

```diff
@@ -128,30 +128,33 @@ void ActiveTcpSocket::continueFilterChain(bool success) {
           no_error = false;
           break;
         } else {
-          // If the listener maxReadBytes() is 0, then it shouldn't return
-          // `FilterStatus::StopIteration` from `onAccept` to wait for more data.
-          ASSERT((*iter_)->maxReadBytes() != 0);
-          if (listener_filter_buffer_ == nullptr) {
-            if ((*iter_)->maxReadBytes() > 0) {
+          // There are two cases for returning StopIteration. One is listener filter
+          // needs to wait for data. The other is listener filter with 0 maxReadBytes()
+          // doesn't inspect data, but needs to wait for some asynchronous callback.
+          if ((*iter_)->maxReadBytes() > 0) {
+            if (listener_filter_buffer_ == nullptr) {
               createListenerFilterBuffer();
```

Comment:
Also, if we don't create ListenerFilterbuffer, then there won't have any notification if the connection closed.

Reply:
I was thinking since listener filter with maxReadBytes() = 0 doesn't interact with listener filter buffer, then the only close notification can be from custom listener filter actively calling socket_->ioHandle().close()

Reply:
There also can be a remote close, if we don't create ListenerFilterBuffer, we won't receive any event about the remote close.

Reply:
Moved createListenerFilterBuffer out so that buffer will be created the first time StopIteration is returned